enhancements/microshift/etcd-as-transient-systemd-unit.md

```diff
@@ -0,0 +1,123 @@
+---
+title: etcd
+authors:
+  - dusk125
+  - hasbro17
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# etcd as a Transient systemd Unit
+
+## Summary
+
+The enhancement proposes moving etcd out of the MicroShift binary and into a transient systemd unit that is launched and managed by MicroShift.
+
+## Motivation
+
+Currently MicroShift [bundles upstream etcd](https://github.com/openshift/microshift/blob/f7f2260e4fbff61654f478fc149cb7052261f87f/go.mod#L21) into its binary and runs it as a goroutine. This results in etcd dependencies being coupled with MicroShift which makes it harder to maintain and upgrade etcd independently of the platform like it is done with a separate openshift/etcd repo in the Openshift Container Platform (OCP).
+
+This enhancement outlines an update to MicroShift that would decouple its deployment of etcd to bring it closer to how etcd is maintained with OCP.
+
+Changing the deployment of etcd, outlined below, would alleviate concerns about building etcd with a version of shared dependencies - mainly library-go - that etcd is not currently built against, and therefore, not tested against.
+
+Additionally moving the etcd deployment into its own process would make it easier to debug and process logs via journatlctl.
+
+### User Stories
+
+1. "As a developer, I want to maintain etcd versions independently of microshift dependencies."
+
+2. "As an admin, I want to observe and debug the etcd server logs separately from other microshift processes."
+
+### Goals
+
+- The etcd binary is built and shipped separately from the microshift binary to ease maintainence
+- The etcd server logs are easier to observe
+
+### Non-Goals
+
+- 
+
+## Proposal
```

Comment:
it's not clear to me. Have we considered shipping etcd as a static pod whose manifest is shipped in the microshift rpm? That seems closest to how openshift runs: underneath cri-o, but managed by an external entity (openshift is etcd-operator, microshift is microshift rpm). Since the systemd-unit would still be shipped by the microshift rpm (right), we get the same lifecycle between the two implementations afaict (to upgrade etcd, upgrade microshift)

Reply:
A static pod was discussed, but not considered very seriously. We want the runtime lifecycle to be bound together, too, not just the upgrade. With the current implementation, if a user stops MicroShift the etcd database is also stopped and it's safe to do things like back it up or wipe it. If we split the 2 processes completely we lose that, so we're trying to find a middle ground.

Reply:
@dusk125 I don't see static pods mentioned in the latest draft as an alternative. Could you add some text based on this thread?